src/components/siteContent/SiteContent.js

```diff
@@ -1,21 +1,56 @@
 import Modal from "components/modal/Modal";
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import styles from "components/siteContent/siteContent.module.css";
 import PropTypes from "prop-types";
-const SiteContent = ({ groups, style }) => {
+const SiteContent = ({ groups }) => {
   const [selectedGroup, setSelectedGroup] = useState(null);
   const [selectedPhotoIndex, setSelectedPhotoIndex] = useState(null);
   const [isModalOpened, setIsModalOpened] = useState(false);
   const [isDescriptionOpened, setIsDescriptionOpened] = useState(false);
   const [hoveredPhotoIndex, setHoveredPhotoIndex] = useState(null);
+  const [isMobile, setIsMobile] = useState(false);
+
+  useEffect(() => {
+    window.scrollTo(0, 0);
+    setIsMobile(!(document.documentElement.clientWidth > 640));
+  }, []);
+
+  const setMargin = groupSize => {
+    if (groupSize === 1) {
+      return isMobile ? "0" : "35%";
+    } else if (groupSize === 2) {
+      return isMobile ? "0" : "22%";
+    } else if (groupSize === 3 || 6) {
+      return isMobile ? "0" : "5%";
+    } else {
+      return "0";
+    }
+  };
+
+  const setGridTemplate = groupSize => {
+    if (groupSize === 1) {
+      return isMobile ? "repeat(1, 1fr)" : "repeat(1, 1fr)";
```

Comment:
tutaj tak samo, sprawdzasz to isMobile w każdy ifie i wszedzie jest taki samy return gdyby isMobile bylo true